### PR TITLE
Add adapter for validation translations

### PR DESCRIPTION
**Pull request by wout**
This PR adds the possibility to configure a custom i18n backed for validation translations, as discussed in https://github.com/luckyframework/avram/issues/754. It also adds simple interpolation using format strings to create clearer error messages. For example in `validate_size_of`, the error message for `min` used to be `"is too short"`, and is now `"must have at least %s characters"`.

### CODE REVIEW

File: src/avram/migrator/runner.cr [hunk 1]
```diff
@@ -197,11 +197,15 @@ class Avram::Migrator::Runner
   end
 
   private def migrated_migrations
-    @@migrations.select &.new.migrated?
+    sorted_migrations.select &.new.migrated?
```
[the remark this answers is not in the document]
Comment (wout) [reply src/avram/migrator/runner.cr:200]: Yeah, sorry abou that. :blush:

File: src/avram/i18n_backend.cr [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      validate_required:           "is required",
+      validate_acceptance_of:      "must be accepted",
+      validate_confirmation_of:    "must match",
+      validate_inclusion_of:       "is not included in the list",
+      validate_exact_size_of:      "must be exactly %s characters long",
+      validate_min_size_of:        "must have at least %s characters",
+      validate_max_size_of:        "must not have more than %s characters",
+      validate_numeric_nil:        "must not be nil",
+      validate_numeric_min:        "must be greater than %s",
+      validate_numeric_max:        "must be less than %s",
```
[the remark this answers is not in the document]
Comment (wout) [reply src/avram/i18n_backend.cr:19]: Good point! I'll fix that.

File: spec/support/i18n_backend_helpers.cr [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module I18nBackendHelpers
+  private def with_i18n_backend(backend : Avram::I18nBackend)
+    Avram.configure do |settings|
+      settings.i18n_backend = backend
+    end
+
+    yield
+  ensure
+    Avram.configure do |settings|
+      settings.i18n_backend = Avram::I18n.new
```
[the remark this answers is not in the document]
Comment (wout) [reply spec/support/i18n_backend_helpers.cr:10]: Aah, I didn't know about `temp_config`. That's a great feature actually.

File: src/avram/migrator/runner.cr (hunk 1 above)
[the remark this answers is not in the document]
Comment (wout) [reply src/avram/migrator/runner.cr:200]: I think I know what happened here. The sorted migrations PR was merged before I fetched upstream. So those changes were already in master, but still, they're shown here? Or am I wrong?

Not that it really matters, it's for my own sanity. I recall branching from master, and not from the sorted migrations branch.

[the remark this answers is not in the document]
Comment (jwoertink) [reply src/avram/migrator/runner.cr:200]: Yeah, you probably branched off before I made the merge or something. It's all good though.

**Review by jwoertink — APPROVED** [review]

**Comment by jwoertink** [comment]
Awesome! This looks great to me. I'll merge it in so we can start playing with it. Thanks!

**Pull request merged**